image-viewer.py

```diff
@@ -1,37 +1,40 @@
-from tkinter import Tk, Label, Button
+from tkinter import Tk, Label, Button, Menu, filedialog
 from PIL import Image, ImageTk
 from os import listdir
 from os.path import isfile
 
 class ImageViewer():
     def __init__(self):
+                     
         self.current_image_index = 0
         self.image_files = []
+        self.blank_img = Image.new('RGBA',(500,300),(0,0,0,0)) # Blank Image to display if no other images are found   
+        self.img_dir = "./images" # Directory to search images in -- by default is the ./images folder
 
-        for item in listdir("./images"):
-            if isfile(f"./images/{item}"):
-                self.image_files.append(item)
-
+        # Tkinter elements setup
         self.root = Tk()
-        self.name_label = Label(self.root, text=self.image_files[self.current_image_index], bg="#1D1D1D", fg="#FFFFFF", font=("Arial 18"))
-        self.current_image = ImageTk.PhotoImage(Image.open(f"./images/{self.image_files[self.current_image_index]}"))
+        self.name_label = Label(self.root, text="No image found", bg="#1D1D1D", fg="#FFFFFF", font=("Arial 18"))
+        self.current_image = ImageTk.PhotoImage(self.blank_img)
         self.display = Label(image=self.current_image, width=500, height=300, bg="#1D1D1D")
         self.move_left_button = Button(self.root, text="<", font=("Calibri 20"), bg="#1C1C1C", fg="#FFFFFF", command=self.move_left)
         self.move_right_button = Button(self.root, text=">", font=("Calibri 20"), bg="#1C1C1C", fg="#FFFFFF", command=self.move_right)
+                
+        # Menu creation
+        self.create_menu()
+        
+        # Refresh the current directory and display images
+        self.refresh()
+        
 
     def move_left(self):
-        if self.current_image_index != 0:
+        if self.current_image_index != 0 and len(self.image_files) != 0:
```

Comment:
I'm probably missing something, but what is `and len(self.image_files) != 0` preventing? If there are 0 files, won't the `current_image_index` always be 0?

Reply:
Likewise on line 35.